pdr_backend/lake/table_bronze_pdr_slots.py

```diff
@@ -54,46 +54,50 @@ def get_bronze_pdr_slots_data_with_SQL(
-                {pdr_slots_table_name}.timestamp as timestamp,
+            SELECT DISTINCT
+                {pdr_slots_table_name}.ID AS ID,
+                SPLIT_PART({pdr_slots_table_name}.ID, '-', 1) AS contract,
+                {etl_bronze_pdr_predictions_table_name}.pair AS pair,
+                {etl_bronze_pdr_predictions_table_name}.timeframe AS timeframe,
+                {etl_bronze_pdr_predictions_table_name}.source AS source,
+                {pdr_slots_table_name}.slot AS slot,
+                {pdr_payouts_table_name}.roundSumStakesUp AS roundSumStakesUp,
+                {pdr_payouts_table_name}.roundSumStakes AS roundSumStakes,
+                {pdr_truevals_table_name}.truevalue AS truevalue,
+                {pdr_slots_table_name}.timestamp AS timestamp,
                 GREATEST(
                     {pdr_slots_table_name}.timestamp,
-                    filtered_{pdr_payouts_table_name}.timestamp,
-                    filtered_{etl_bronze_pdr_predictions_table_name}.timestamp,
-                    {pdr_truevals_table_name}.timestamp
-                ) as last_event_timestamp,
+                    {pdr_truevals_table_name}.timestamp,
+                    {pdr_payouts_table_name}.timestamp,
+                    {etl_bronze_pdr_predictions_table_name}.timestamp
+                ) AS last_event_timestamp
             FROM
                 {pdr_slots_table_name}
-            LEFT JOIN {pdr_truevals_table_name}
-            ON {pdr_slots_table_name}.ID = {pdr_truevals_table_name}.ID
+            LEFT JOIN {pdr_truevals_table_name} ON {pdr_slots_table_name}.ID = {pdr_truevals_table_name}.ID
             LEFT JOIN (
-                SELECT *,
-                    SPLIT_PART({pdr_payouts_table_name}.ID, '-', 1)
-                || '-' || SPLIT_PART({pdr_payouts_table_name}.ID, '-', 2) as match_id,
-                ROW_NUMBER() OVER (PARTITION BY match_id) as rn,
-                FROM {pdr_payouts_table_name} 
-            ) AS filtered_{pdr_payouts_table_name} 
-            ON {pdr_slots_table_name}.ID = filtered_{pdr_payouts_table_name}.match_id
-            AND filtered_{pdr_payouts_table_name}.rn = 1
+                SELECT 
+                    {pdr_slots_table_name}.ID AS slot_id,
+                    MAX({etl_bronze_pdr_predictions_table_name}.pair) AS pair,
+                    MAX({etl_bronze_pdr_predictions_table_name}.timeframe) AS timeframe,
+                    MAX({etl_bronze_pdr_predictions_table_name}.source) AS source,
+                    MAX({etl_bronze_pdr_predictions_table_name}.timestamp) AS timestamp
+                FROM {pdr_slots_table_name}
+                LEFT JOIN {etl_bronze_pdr_predictions_table_name} ON SPLIT_PART({pdr_slots_table_name}.ID, '-', 1) = {etl_bronze_pdr_predictions_table_name}.contract
+                GROUP BY {pdr_slots_table_name}.ID
+            ) AS {etl_bronze_pdr_predictions_table_name} ON {pdr_slots_table_name}.ID = {etl_bronze_pdr_predictions_table_name}.slot_id
```

Comment:
we should not use a JOIN result instead of an existing table's name, here `{etl_bronze_pdr_predictions_table_name}`. I think we have to use an alias.